authz/audit/stdout/stdout_logger_test.go

```diff
@@ -62,26 +62,29 @@ func (s) TestStdoutLogger_Log(t *testing.T) {
 		},
 	}
 
-	content := json.RawMessage(`{"name": "conf", "val": "to be ignored"}`)
-	builder := &loggerBuilder{}
-	config, _ := builder.ParseLoggerConfig(content)
-	auditLogger := builder.Build(config)
-	log.SetFlags(0)
-
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
+			before := time.Now().Unix()
 			var buf bytes.Buffer
-			log.SetOutput(&buf)
+			l := log.New(&buf, "", 0)
+			auditLogger := &logger{
+				goLogger: l,
+			}
 			auditLogger.Log(test.event)
-			var e event
-			if err := json.Unmarshal(buf.Bytes(), &e); err != nil {
+			var container map[string]interface{}
+			if err := json.Unmarshal(buf.Bytes(), &container); err != nil {
 				t.Fatalf("Failed to unmarshal audit log event: %v", err)
 			}
-			if strings.TrimSpace(e.Timestamp) == "" {
-				t.Fatalf("Resulted event has no timestamp: %v", e)
+			innerEvent := extractEvent(container["grpc_audit_log"].(map[string]interface{}))
+			if innerEvent.Timestamp == 0 {
+				t.Fatalf("Resulted event has no timestamp: %v", innerEvent)
 			}
-
-			if diff := cmp.Diff(trimEvent(e), test.event); diff != "" {
+			after := time.Now().Unix()
+			if before > innerEvent.Timestamp || after < innerEvent.Timestamp {
+				t.Fatalf("The audit event timestamp is outside of the test interval: test start %v, event timestamp %v, test end %v",
+					before, innerEvent.Timestamp, after)
+			}
+			if diff := cmp.Diff(trimEvent(innerEvent), test.event); diff != "" {
```

Comment:
As we talked offline, I suggest comparing the values of the same type but ignoring `timestamp` via `cmpopts.IgnoreFields`.

Reply:
After further offline discussion, this is fine as is.